### PR TITLE
Redis no auth

### PR DESCRIPTION
**Pull request by Richard-Mathie**
Add option to have no auth token when connecting to redis

rational:

- aws elasticache does not support auth tokens without ssl
- ssl requires a truststore? ect to be set up

is this an ideal solution? no
did it work: yes
am i proud: no

notes:
the SECRET_I_DONT_MATTER envar to make the lense-cli create the secret properties file. This is a hack.

next steps is to try and get redis and ssl working.

### CODE REVIEW

File: charts/kafka-connect-redis-sink/templates/deployment.yaml [hunk 1]
```diff
@@ -214,23 +214,30 @@ spec:
...
+        {{- if .Values.password }}
+        # only create SECRET_CONNECT_REDIS_PASSWORD if auth token supplied
         - name: SECRET_CONNECT_REDIS_PASSWORD
           valueFrom:
             secretKeyRef:
               name: {{ include "fullname" . }}
               key:  "connect.redis.password"
+        {{- else }}
+        # Hack so that lenses-cli will create /etc/kafka-connect/secrets.properties
+        - name: SECRET_I_DONT_MATTER
```
Comment (spirosoik) [inline charts/kafka-connect-redis-sink/templates/deployment.yaml]: not the best solution

Comment (andrewstevenson) [reply charts/kafka-connect-redis-sink/templates/deployment.yaml]: @spirosoik We should apply a fix in the lenses-cli if we need it

**Comment by spirosoik** [comment]
closing this PR as there is no need to change this part. more details in https://github.com/Landoop/kafka-helm-charts/pull/100

File: charts/kafka-connect-redis-sink/templates/deployment.yaml (hunk 1 above)
Comment (Richard-Mathie) [reply charts/kafka-connect-redis-sink/templates/deployment.yaml]: @spirosoik lol yes i know....

as an aside why is it we are not just injecting the .properties files directly in the helm chart?

**Comment by Richard-Mathie** [comment]
Not sure if #100 fixes as the aws elasticache is just requiring the connection to tls upgrade, so I dont think it uses a trust certificate. that is I think it just tls encrypts the traffic (I tested with stunnel and telnet) but does not protect against MITM. so yeah not ideal, but all traffic should be within your VPC...

**Comment by spirosoik** [comment]
Some context:
 ```ElastiCache for Redis takes care of certificate issuance process transparently to your application. You do not need to acquire a certificate from a certificate authority and deploy/upload the certificate once acquired. ElastiCache for Redis uses a trusted Certificate Authority behind the scenes, minimizing any client set up or costs associated with a third-party certificate authority.

As long as the Redis clients support TLS protocol, you don't have to modify your applications other than making a small configuration change to your Redis clients. The change to your Redis client is simple, for example, with a Jedis java client, while creating a Redis connection, you need to pass isTls=true flag.
```

This is by AWS official docmentation for Encrypt data in transit https://aws.amazon.com/about-aws/whats-new/2017/10/amazon-elasticache-for-redis-now-supports-in-transit-and-at-rest-encryption-to-help-protect-sensitive-information/

**Comment by Richard-Mathie** [comment]
yeah i saw that, also this can be picked up automatically by the jedis client by using a protocol of
`rediss://` in th uri see https://github.com/xetorthio/jedis/blob/master/src/test/java/redis/clients/jedis/tests/SSLJedisTest.java#L51

but the way the connector sets up the connection, I think this is lossed.

**Comment by andrewstevenson** [comment]
Jedis lets you specify SSL in the URI or an explicit flag. The connect now explicitly sets the flag.

**Pull request closed without merging**